### PR TITLE
Refactor export instance video script

### PR DESCRIPTION
**Pull request by holger-nutonomy**
This PR improves the instance video script:
- Change the default codec to MPEG.
- Reformatting to align more with rest of the devkit.
- Adds type hints and resolves PEP issues.

Besides that it also fixes a big in a similar script when no out_dir is provided.

### CODE REVIEW

**Comment by holger-nutonomy** [comment]
FYI @EricWiener. I am making some changes to align the script more with the rest of the codebase.

File: python-sdk/nuscenes/scripts/export_instance_videos.py [hunk 1]
```diff
@@ -1,229 +1,245 @@
...
+
+    Parameters:
+    - bounding_box: np.array of length 4 (x min, y min, x max, y max)
+    """
+    x_min, y_min, x_max, y_max = bounding_box
+    return (x_max - x_min) * (y_max - y_min)
+
+
+def get_most_visible_camera_annotation(camera_data_dict: dict) -> dict:
+    """
```
Comment (kokseang-aptiv) [inline python-sdk/nuscenes/scripts/export_instance_videos.py:119]: Docstring?

File: python-sdk/nuscenes/scripts/export_instance_videos.py [hunk 2]
```diff
@@ -255,21 +281,25 @@ def main(version, dataroot, output, object_categories, fps, output_size, minimum
...
         maps	-	Folder for all map files: rasterized .png images and vectorized .json files.
-        v1.0-*	-	JSON tables that include all the meta data and annotations. Each split (trainval, test, mini) is provided in a separate folder.
+        v1.0-*	-	JSON tables that include all the meta data and annotations.
+                    Each split (trainval, test, mini) is provided in a separate folder.
                     Note that image_annotations.json should be inside this directory.
     ```
     
     Args:
-        version (string): the NuScenes data version
+        version (string): the nuScenes data version
```
Comment (kokseang-aptiv) [inline python-sdk/nuscenes/scripts/export_instance_videos.py:290]: Upper case of the first letters and full stop?

File: python-sdk/nuscenes/scripts/export_instance_videos.py [hunk 3]
```diff
@@ -1,229 +1,245 @@
...
+
+        sequential_sample_tokens.append(next_sample_token)
+
+    return sequential_sample_tokens
+
+
+def remove_bad_samples(instance_annotation: dict,
+                       minimum_bb_area: float,
+                       minimum_visibility: str,
+                       image_area=1600 * 900) -> List:
```
Comment (kokseang-aptiv) [inline python-sdk/nuscenes/scripts/export_instance_videos.py:242]: Type for image_area, which is int?

**Review by kokseang-aptiv — APPROVED** [review]
Looks good. Few minor comments only.

File: python-sdk/nuscenes/scripts/export_instance_videos.py [hunk 4]
```diff
@@ -1,229 +1,245 @@
...
-      next_sample_token = instance_dict[sequential_sample_tokens[-1]
-                                        ]['best_annotation']['next']
-    except:
-      print("Unrecognized sample annotaton token: {}", sequential_sample_tokens)
-      break
+    camera_name = filename.split('/')[1]
 
-    if next_sample_token == '':
-      break
+    # Validate the camera name is valid
```
Comment (whyekit-aptiv) [inline python-sdk/nuscenes/scripts/export_instance_videos.py]: Full stop at end of sentence?

File: python-sdk/nuscenes/scripts/export_instance_videos.py [hunk 5]
```diff
@@ -1,229 +1,245 @@
...
+    bbox = sample_data_annotation['bbox_corners']
+    im = Image.open(data_path)
+    im1 = im.crop(bbox)
+    im1 = im1.resize(output_size)
+    np_img = np.asarray(im1)
+    return np_img
+
+
+def sort_sample_annotations_chronologically(instance_dict: dict) -> List[str]:
+    """
```
Comment (whyekit-aptiv) [inline python-sdk/nuscenes/scripts/export_instance_videos.py:210]: Docstring for this?

File: python-sdk/nuscenes/scripts/export_instance_videos.py (hunk 2 above)
Comment (holger-nutonomy) [reply python-sdk/nuscenes/scripts/export_instance_videos.py:290]: nuScenes

File: python-sdk/nuscenes/scripts/export_instance_videos.py [hunk 6]
```diff
@@ -1,238 +1,260 @@
...
+
+        if instance_token in bbox_2d_annotations and sample_token in bbox_2d_annotations[instance_token] \
+                and camera_name in bbox_2d_annotations[instance_token][sample_token]:
+            num_dups += 1
+            print('Duplicate instance {}, sample {}, and camera {}'.format(
+                instance_token, sample_token, camera_name))
+
+        bbox_2d_annotations[instance_token][sample_token][camera_name] = instance
+
+    print("Number of duplicates (should be zero)", num_dups)
```
Comment (whyekit-aptiv) [inline python-sdk/nuscenes/scripts/export_instance_videos.py:90]: If the number of duplicates should be zero, is an assertion better instead?

**Review by whyekit-aptiv — COMMENTED** [review]
Looks good overall! Just a couple of minor comments

**Pull request merged**